### PR TITLE
fix: stop pool connection if credentials fail

### PR DESCRIPTION
**Pull request by abc3**

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -154,7 +154,8 @@ if config_env() != :test do
     prom_poll_rate: System.get_env("PROM_POLL_RATE", "15000") |> String.to_integer(),
     global_upstream_ca: upstream_ca,
     global_downstream_cert: downstream_cert,
-    global_downstream_key: downstream_key
+    global_downstream_key: downstream_key,
+    reconnect_on_db_close: System.get_env("RECONNECT_ON_DB_CLOSE") == "true"
```
Comment (chasers) [inline config/runtime.exs:158]: What's the point of the env var exactly?

Comment (abc3) [reply config/runtime.exs:158]: to configure the behavior of db_handlers when Supervisor loses connection with the tenant's db. It allows to choose between two options: reconnecting to the database (the current behavior) or stopping the db_handler (which will become the default setting). If a large pool of db_handlers attempts to reconnect without valid credentials, the Supervisor IP will be banned

Comment (abc3) [reply config/runtime.exs:158]: https://github.com/supabase/supavisor/blob/c6dc7d27531fdc4fbe5aecb6413e07b09ff02277/lib/supavisor/db_handler.ex#L417-L419

Comment (chasers) [reply config/runtime.exs:158]: Why is it an env var if it's going to be the default is what I mean?

**Comment by chasers** [comment]
I don't really see the reason to make this an env var but if you want to that is fine.

**Review by chasers — APPROVED** [review]

**Pull request merged**